Cringe.Web/Controllers/RootController.cs

```diff
@@ -50,6 +53,31 @@ public async Task<IActionResult> DownloadBeatmap(string id, [FromQuery(Name = "u
             return NotFound();
         }
 
+        [HttpPost("recalculate-scores")]
+        public async Task<IActionResult> RecalculateScores([FromQuery(Name = "securityKey")] string securityKey,
+            [FromServices] ScoreDatabaseContext databaseContext,
+            [FromServices] PpService ppService,
+            [FromServices] IConfiguration configuration)
+        {
+            if (securityKey != configuration["SecurityKey"]) return Unauthorized();
+
+            _logger.LogWarning("Started PP recalculation");
+
+            var submittedScores = await databaseContext.RecentScores.ToListAsync();
+
+            foreach (var recentScore in submittedScores)
+            {
+                _logger.LogInformation("Recalculation: {ScoreId}", recentScore.Id);
+                var newPp = await ppService.CalculatePp(recentScore);
+                _logger.LogInformation("PP change {OldPp} -> {NewPp}", recentScore.Pp, newPp);
+                recentScore.Pp = newPp;
```

Comment:
`await databaseContext.Update(recentScore)`

Reply:
Unnecessary and lead to worse performance overall, see: https://learn.microsoft.com/en-us/ef/core/change-tracking/